### PR TITLE
Wrap up documentation for partial transformers

### PR DESCRIPTION
**Pull request by MateuszKubuszok**
TODO:

- [x] document the new flag `enableImplicitConflictResolution`
- [x] link to the docs in the flag scaladoc (remove TODOs)
- [x] partial example in index.rs

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -55,9 +55,88 @@ rules and transform your objects with as little boilerplate as possible.
...
 
+Partial transformers
+--------------------
+
+Examples so far described situation when every value of one type can be converted into another type. But what when only
+some values can be converted?
+
+.. code-block:: scala
+
+  case class ApiUser(name: String, ageInput: String, email: Option[String])
```
Comment (krzemin) [inline docs/source/index.rst:69]: naming: `UserForm`?

File: docs/source/index.rst [hunk 2]
```diff
@@ -55,9 +55,88 @@ rules and transform your objects with as little boilerplate as possible.
...
+alternative:
+
+.. code-block:: scala
+
+  import io.scalaland.chimney._
+  import io.scalaland.chimney.dsl._
+  import io.scalaland.chimney.partial._
+
+  val success = ApiUser("John", "21", Some("john@example.com")).intoPartial[User]
+    .withFieldComputedPartial(_.age, api => Result.fromCatching(api.ageInput.toInt))
```
Comment (krzemin) [inline docs/source/index.rst:85]: `userForm =>`

File: docs/source/index.rst [hunk 3]
```diff
@@ -55,9 +55,88 @@ rules and transform your objects with as little boilerplate as possible.
...
+    .transform
+ val failure = ApiUser("Ted", "eighteen", None).intoPartial[User]
+    .withFieldComputedPartial(_.age, api => Result.fromCatching(api.ageInput.toInt))
```
Comment (krzemin) [inline docs/source/index.rst:88]: `userForm =>`?

File: docs/source/index.rst [hunk 4]
```diff
@@ -55,9 +55,88 @@ rules and transform your objects with as little boilerplate as possible.
   // CoffeeMade(24, "Espresso", "Piotr", "2020-02-03T20:26:59.659647+07:00[Asia/Bangkok]")
 
 
-Read :ref:`transformers/getting-started:Getting started with transformers` to learn more about
+Read :ref:`partial-transformers/getting-started:Getting started with transformers` to learn more about
 Chimney's transformers.
 
+Partial transformers
```
Comment (krzemin) [inline docs/source/index.rst:61]: Seems we have too many, too detailed code examples in this section.

That shouldn't be in-depth explanation what you can do with partial transformers (we have dedicated page for that!), but rather introductory _catch-the-idea_ example that signals that we support transformers that can fail.

The same simple example should be used in `README.md`, I guess?

File: docs/source/partial-transformers/partial-transformers.rst [hunk 5]
```diff
@@ -166,6 +166,54 @@ You can expect that basic functionality of chimney's ``Transformer`` either work
...
+
+  case class UserAPI(credits: String)
+  case class User(credits: Int)
+
+  implicit val unsafeStringToInt: Transformer[String, Int] = _.toInt
+
+  implicit val parseStringToInt: PartialTransformer[String, Int] =
+    PartialTransformer[String, Int](str => partial.Result.fromCatching(str.toInt).map(_ * 2))
+
+  UserAPI("10").transformIntoPartial[User].asOption // Some(User(10)) or Some(User(20)) ?
```
Comment (krzemin) [inline docs/source/partial-transformers/partial-transformers.rst:191]: We should explicitly mention that this example ends up with a compilation error.

File: docs/source/partial-transformers/partial-transformers.rst [hunk 6]
```diff
@@ -166,6 +166,54 @@ You can expect that basic functionality of chimney's ``Transformer`` either work
...
 (usually with the `Partial` prefix or suffix in the name).
 
+Preferring total or partial transformer
+---------------------------------------
+
+Under normal circumstances when fields or sealed hierarchy subtypes should be converted, Chimney should be able
+to generate code on its own. If it cannot you could provide it with derivation settings or implicit transformer handling
+problematic fields/subtypes.
+
+But which implicit should be taken in this situation?
```
Comment (krzemin) [inline docs/source/partial-transformers/partial-transformers.rst:176]: How about non-ambiguous examples, when we have just implicit `Transformer` and just implicit `PartialTransformer` in scope?

File: docs/source/partial-transformers/partial-transformers.rst [hunk 7]
```diff
@@ -166,6 +166,54 @@ You can expect that basic functionality of chimney's ``Transformer`` either work
 fashion in ``PartialTransformer``\s, or have some counterparty methods in the API
 (usually with the `Partial` prefix or suffix in the name).
 
+Preferring total or partial transformer
```
Comment (krzemin) [inline docs/source/partial-transformers/partial-transformers.rst:169]: Haven't we considered moving that to a separate page?

**Review by krzemin — CHANGES_REQUESTED** [review]

File: docs/source/index.rst (hunk 2 above)
Comment (MateuszKubuszok) [reply docs/source/index.rst]: done

File: docs/source/partial-transformers/partial-transformers.rst (hunk 5 above)
Comment (MateuszKubuszok) [reply docs/source/partial-transformers/partial-transformers.rst]: done

File: docs/source/partial-transformers/partial-transformers.rst (hunk 7 above)
Comment (MateuszKubuszok) [reply docs/source/partial-transformers/partial-transformers.rst]: done

File: docs/source/partial-transformers/partial-transformers.rst (hunk 6 above)
Comment (MateuszKubuszok) [reply docs/source/partial-transformers/partial-transformers.rst]: done

File: docs/source/index.rst (hunk 3 above)
Comment (MateuszKubuszok) [reply docs/source/index.rst]: done

**Review by krzemin — APPROVED** [review]

**Pull request merged**